iree/compiler/Dialect/HAL/Transforms/InlineDeviceSwitches.cpp

```diff
@@ -76,62 +77,82 @@ static Value buildConditionExpression(Location loc, Value device,
   return {};
 }
 
-// Outlines a condition region from a switch op into a standalone function.
-static FuncOp outlineConditionRegion(StringRef funcName,
-                                     Region &conditionRegion,
-                                     ArrayRef<Type> resultTypes,
-                                     OpBuilder moduleBuilder) {
+// Inlines a condition region from a switch op into the function at the given
+// point. This assumes that the insertion point will only be reached if the
+// condition the region is predicated on is true.
+static void inlineConditionRegion(OperandRange regionArgs,
+                                  Region &conditionRegion, Block *exitBlock,
+                                  OpBuilder funcBuilder) {
+  assert(!conditionRegion.empty() && "source regions must not be empty");
+
+  // Split the insertion block.
+  auto *insertBlock = funcBuilder.getBlock();
+  auto *postInsertBlock = &*std::next(insertBlock->getIterator());
```

Comment:
I'm confused. This doesn't seem to split a block. The goal here is to add in new blocks immediately after the current insertion point, right? It seems like there must be a more idiomatic way to do that... I think part of the confusion comes from this being somewhat far away from the place where you're actually doing the splicing and the corresponding iterator manipulation. Also because there's a method called "splitBlock" which you're not using

Reply:
If you go invent a more idiomatic way then you can update this code to use it :shrug:.
I don't particularly consider what InliningUtils (which is rather old at this point, MLIR-wise) to be the most up to date code, but I frankly don't really care to differ from it for the sake of being different.

If by "somewhat far away" you mean 10 lines of code in the same function I think you should decrease your font size ;) I'm not using splitBlock because I'm not splitting blocks: I'm inserting things into a split between blocks that is created explicitly for this function to use.

Reply:
Right I understand why you're not using split block, but it's weird to have a comment that says "split the block" and then not have it use the function "splitBlock". Basically, the existence of that function has defined what splitting a block means and that's not what you're doing here.

If you're going to copy-paste code from somewhere else and want to just leave it the same, please put a comment to that affect so I don't waste my time reviewing it.